sky/optimizer.py

```diff
@@ -826,7 +825,6 @@ def _fill_in_launchable_resources(
     try_fix_with_sky_check: bool = True,
 ) -> Tuple[Dict[resources_lib.Resources, List[resources_lib.Resources]],
            _PerCloudCandidates]:
-    backend_utils.check_public_cloud_enabled()
```

Comment:
@Michaelvll Do you remember the context when this line of code is added? Correct me if I'm wrong, but `backend_utils.check_public_cloud_enabled()` seems to be redundant as it is checked in `cli.py`.

Reply:
Perhaps it was added for the programmatic API?